File: feeder-webapp/src/main/java/com/labsigma/feederwebapp/entities/Feeder.java
package com.labsigma.feederwebapp.entities;

import lombok.Data;

import java.util.ArrayList;
import java.util.List;

@Data
public class Feeder {
    private Long id;
    private String description;
    private Float longitude;
    private Float latitude;
    private List<SensorMeasurement> sensorMeasurements;
    private List<BirdFile> birdFiles;

    public Feeder() {
        sensorMeasurements = new ArrayList<>();
        birdFiles = new ArrayList<>();
    }
}

File: feeder-webapp/front/src/app/interfaces/feeder.ts
import {SensorMeasurement} from './sensor-measurement';
import {BirdFile} from './bird-file';

export interface Feeder {
  id: number;
  description: string;
  longitude: number;
  latitude: number;
  sensorMeasurements: SensorMeasurement[];
  birdFiles: BirdFile[];
}
